src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -0,0 +1,148 @@
+
+
+  // constructor
+  public Shooter() {
+
+    // instantiate objects
+    leftShooterMotor = new TalonFX(leftShooterMotorID);
+    rightShooterMotor = new TalonFX(rightShooterMotorID);
+    loaderMotor = new TalonFX(loaderMotorID);
+
+    inputSensor = new DigitalInput(3);
+
+    slotConfigsR = new Slot0Configs();
+    slotConfigsR.kS = rShooterMotorSGains;
+    slotConfigsR.kV = rShooterMotorVGains;
+    slotConfigsR.kP = rShooterMotorPGains; 
+    slotConfigsR.kI = rShooterMotorIGains;
+    slotConfigsR.kD = rShooterMotorDGains;
+
+
+    slotConfigsL = new Slot0Configs();
+    slotConfigsL.kS = lShooterMotorSGains;
+    slotConfigsL.kV = lShooterMotorVGains;
+    slotConfigsL.kP = lShooterMotorPGains;
+    slotConfigsL.kI = lShooterMotorIGains;
+    slotConfigsL.kD = lShooterMotorDGains;
+
+    rm_request = new VelocityVoltage(0).withSlot(0);
+    lm_request = new VelocityVoltage(0);
+
+    leftShooterMotor.setNeutralMode(NeutralModeValue.Coast);
+    rightShooterMotor.setNeutralMode(NeutralModeValue.Coast);
+
+    SmartDashboard.putNumber("RShooter kP", slotConfigsR.kP);
+    SmartDashboard.putNumber("RShooter kV", slotConfigsR.kV);
+    SmartDashboard.putNumber("RShooter Speed", 0);
+    SmartDashboard.putNumber("LShooter kP", slotConfigsL.kP);
+    SmartDashboard.putNumber("LShooter kV", slotConfigsL.kV);
+    SmartDashboard.putNumber("LShooter Speed", 0);
+    shootingMotorsConfig();
+  }
+
+  public void runLoader() {
+    loaderMotor.setVoltage(12.0);  
+  }
+
+    public void reverseLoader() {
+    loaderMotor.setVoltage(-3.0);  
+  }
+
+  public void stopLoader() {
+    loaderMotor.setVoltage(0);
+  }
+
+  public boolean sensorValue() {
+    return inputSensor.get();
+  }
+
+  public void shootingMotorsConfig() {
+    slotConfigsR.kP = SmartDashboard.getNumber("RShooter kP", 0);
+    slotConfigsR.kV = SmartDashboard.getNumber("RShooter kV", 0);
+    slotConfigsL.kP = SmartDashboard.getNumber("LShooter kP", 0);
+    slotConfigsL.kV = SmartDashboard.getNumber("LShooter kV", 0);
+    rightShooterMotor.getConfigurator().apply(slotConfigsR);
+    leftShooterMotor.getConfigurator().apply(slotConfigsL);
+  }
+
+  public void shootingMotorsSetControl(double rightShooterSpeed, double leftShooterSpeed) {
+
+    rightShooterMotor.setControl(rm_request.withVelocity(rightShooterSpeed));
+    leftShooterMotor.setControl(lm_request.withVelocity(leftShooterSpeed));
+
+  }
+
+  public void setShooterVoltage(double rightVoltage, double leftVoltage) {
+    rightShooterMotor.setVoltage(rightVoltage);
+    leftShooterMotor.setVoltage(leftVoltage);
+  }
+
+  public void stop() {
+    rightShooterMotor.setControl(rm_request.withVelocity(0));
+    leftShooterMotor.setControl(rm_request.withVelocity(0));
+  }
+
+
+  
+
+  @Override
+  public void periodic() {
+    // This method will be called once per scheduler run    
+    SmartDashboard.putNumber("Right Shooter Speed", rightShooterMotor.getVelocity().getValueAsDouble());
```

Comment:
We should be careful with what data requests we are putting in any subsystem periodic. Make sure it is part of the status frame of that motor. I can't find the status frames for a TalonFX but I did link them for a SPARK MAX below. Talk to me if you don't know what a status frame is. 

https://docs.revrobotics.com/brushless/spark-max/control-interfaces#periodic-status-frames